lib/logproto/logproto-server.c

```diff
@@ -106,10 +106,21 @@ log_proto_server_validate_options_method(LogProtoServer *s)
   return TRUE;
 }
 
+void
+log_proto_server_close_connection(LogProtoServer *s)
+{
+  if (s->transport)
+    {
+      log_transport_free(s->transport);
+      s->transport = NULL;
+    }
+}
+
 void
 log_proto_server_free_method(LogProtoServer *s)
 {
-  log_transport_free(s->transport);
+  if (s->transport)
+    log_transport_free(s->transport);
```

Comment:
I think free should set `transport` to `NULL`, just in case.

Reply:
The entire `LogProtoServer` is destroyed after calling this function, but I can add the NULL assignment to keep the symmetry with `close_connection` and just to be sure.